src/java_bytecode/jar_file.cpp

```diff
@@ -96,16 +138,17 @@ std::string jar_filet::get_entry(std::size_t i)
 
   std::string dest;
 
+  size_t real_index=filtered_jar[i];
```

Comment:
The simpler the better, I'd say. This is not a code path where performance matters as much, it's mostly about keeping it free from bugs...